src/adapters/default/config.adapter.ts

```diff
@@ -0,0 +1,155 @@
+    let data = {
+      filters: {
+        module: {
+          eq: `${configDto.module}`,
+        },
+        key: {
+          eq: `${configDto.key}`,
+        },
+        contextId: {
+          eq: `${configDto.contextId}`,
+        },
+      },
+    };
+
+    let config = {
+      method: "post",
+      url: `${this.url}/search`,
+
+      data: data,
+    };
+
+    const response = await axios(config);
+    let resData = response?.data;
+    let result = resData.map((item: any) => new ConfigDto(item));
+
+    let configId = result.map(function (ConfigDto) {
+      return ConfigDto.configId
+    })
+
+    if (resData.length > 0)
+    {
+      var udateData = configDto;
+      var updateConfig = {
+      method: "put",
+      url: `${this.url}/${configId}`,
+      headers: {
+        Authorization: request.headers.authorization,
+      },
+      data: udateData,
+      };
+      const response = await axios(updateConfig);
+      console.log("123");
+      console.log(configId);
+      console.log(response);
+      return new SuccessResponse({
+        statusCode: 200,
+        message: " Ok.",
+        data: response.data,
+      });
+    }
+    else
+    {
+      return this.httpService
+      .post(`${this.url}`, configDto, {
+        headers: {
+          Authorization: request.headers.authorization,
+        },
+      })
+      .pipe(
+        map((axiosResponse: AxiosResponse) => {
+          console.log("140" + axiosResponse.data);
+          return new SuccessResponse({
+            statusCode: 200,
+            message: "Ok.",
+            data: axiosResponse.data,
+          });
+        }),
+        catchError((e) => {
+          var error = new ErrorResponse({
+            errorCode: e.response?.status,
+            errorMessage: e.response?.data?.params?.errmsg,
+          });
+          throw new HttpException(error, e.response.status);
+        })
+      );
+    }
+  }
+
+  public async getConfigForTeacher(request: any) {
+    const authToken = request.headers.authorization;
+    const decoded: any = jwt_decode(authToken);
+    let email = decoded.email;
+
+    let axios = require("axios");
+    let teacherData = {
+      filters: {
+        email: {
+          eq: `${email}`,
+        },
+      },
+    };
+    let config = {
+      method: "post",
+      url: `${process.env.BASEAPIURL}/Teacher/search`,
+      headers: {
+        Authorization: request.headers.authorization,
+      },
+      data: teacherData,
+    };
+    const response = await axios(config);
+
+    let teacherProfileData =
+      response?.data && response.data.map((item: any) => new TeacherDto(item));
+      
+      let schoolId = teacherProfileData.map(function (TeacherDto) {
+        return TeacherDto.schoolId
+      })
+
+      let data = {
+        filters: {
+          contextId: {
+            eq: `${schoolId}`,
+          },
+        },
+      };
+
+      let final = {
+        method: "post",
+        url: `${this.url}/search`,
```

Comment:
If the school specific config is not available it should use the global one. This does not seem to be implemented ?

Reply:
Sent global config, if School specific not found